core/primitives/src/types.rs

```diff
@@ -1093,6 +1093,13 @@ pub trait EpochInfoProvider {
 
     /// Get the chain_id of the chain this epoch belongs to
     fn chain_id(&self) -> String;
+
+    /// Which shard the account belongs to in the given epoch.
+    fn account_id_to_shard_id(
+        &self,
+        account_id: &AccountId,
+        epoch_id: &EpochId,
+    ) -> Result<ShardId, EpochError>;
```

Comment:
Generally speaking yes. We decided to use shard_ids as it would require a major refactoring to do it right. Somewhat related in the trie and headers we actually use u16 in order to reduce the overhead.